tests/src/whisk/core/database/test/CacheConcurrencyTests.scala

```diff
@@ -31,45 +29,112 @@ import common.Wsk
 import common.WskProps
 import whisk.common.SimpleExec
 import whisk.common.TransactionId
+import scala.collection.parallel._
+import scala.concurrent.forkjoin.ForkJoinPool
+import common.TestUtils._
+import spray.json.JsString
+import common.TestHelpers
+import common.WskTestHelpers
 
 @RunWith(classOf[JUnitRunner])
 class CacheConcurrencyTests extends FlatSpec
-    with BeforeAndAfterAll
-    with Matchers {
-
-    private val wsk = new Wsk
+    with WskTestHelpers {
 
-    implicit private val logger = this
     implicit private val transId = TransactionId.testing
     implicit private val wp = WskProps()
+    private val wsk = new Wsk
 
     val nExternalIters = 1
-    val nInternalIters = 10
+    val nInternalIters = 5
+    val nThreads = nInternalIters * 30
+
+    val parallel = (1 to nInternalIters).par
+    parallel.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(nThreads))
 
-    for (i <- 1 to nExternalIters)
-        "the cache" should s"support concurrent CRUD without bogus residual cache entries, iter ${i}" in {
+    def run[W](phase: String)(block: String => W) = parallel.map { i =>
+        val name = s"testy${i}"
+        withClue(s"$phase: failed for $name") { (name, block(name)) }
+    }
+
+    for (n <- 1 to nExternalIters)
+        "the cache" should s"support concurrent CRUD without bogus residual cache entries, iter ${n}" in {
             try {
```

Comment:
Does it make sense to drop the `try` and implement the cleanup in an `afterEach`?